src/main/java/uk/gov/hmcts/reform/pip/account/management/service/MediaLegalApplicationService.java

```diff
@@ -0,0 +1,144 @@
+import uk.gov.hmcts.reform.pip.account.management.database.MediaLegalApplicationRepository;
+import uk.gov.hmcts.reform.pip.account.management.errorhandling.exceptions.NotFoundException;
+import uk.gov.hmcts.reform.pip.account.management.model.MediaAndLegalApplication;
+import uk.gov.hmcts.reform.pip.account.management.model.MediaLegalApplicationStatus;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+public class MediaLegalApplicationService {
+
+    private final MediaLegalApplicationRepository mediaLegalApplicationRepository;
+
+    private final AzureBlobService azureBlobService;
+
+    LocalDateTime now = LocalDateTime.now();
+
+    @Autowired
+    public MediaLegalApplicationService(MediaLegalApplicationRepository mediaLegalApplicationRepository,
+                                        AzureBlobService azureBlobService) {
+        this.mediaLegalApplicationRepository = mediaLegalApplicationRepository;
+        this.azureBlobService = azureBlobService;
+    }
+
+    /**
+     * Get a list of all the applications.
+     *
+     * @return A list of all applications
+     */
+    public List<MediaAndLegalApplication> getApplications() {
+        return mediaLegalApplicationRepository.findAll();
+    }
+
+    /**
+     * Get a list of applications by the status.
+     *
+     * @param status The MediaLegalApplicationStatus enum to retrieve applications by
+     * @return A list of all applications with the relevant status
+     */
+    public List<MediaAndLegalApplication> getApplicationsByStatus(MediaLegalApplicationStatus status) {
+        return mediaLegalApplicationRepository.findByStatus(status);
+    }
+
+    /**
+     * Get an application by the application id.
+     *
+     * @param id The id of the application
+     * @return The application if it exists
+     */
+    public MediaAndLegalApplication getApplicationById(UUID id) {
+        Optional<MediaAndLegalApplication> returnedApplication = mediaLegalApplicationRepository.findById(id);
+        if (returnedApplication.isEmpty()) {
+            throw new NotFoundException(String.format("Application with id %s could not be found", id));
+        }
+
+        return returnedApplication.get();
+    }
+
+    /**
+     * Get an image from the blob store by the image id.
+     *
+     * @param imageId The id of the image to retrieve
+     * @return The file if found
+     */
+    public Resource getImageById(String imageId) {
+        Optional<MediaAndLegalApplication> returnedApplication = mediaLegalApplicationRepository.findByImage(imageId);
+
+        if (returnedApplication.isEmpty() || !returnedApplication.get().getImage().equals(imageId)) {
+            throw new NotFoundException(String.format("Application with image id %s could not be found", imageId));
+        }
+
+        MediaAndLegalApplication application = returnedApplication.get();
```

Comment:
this all seems very unnceccessary based on your method "get image by id" youre actually getting an application first based on a image id then getting the image at the end, 90% of this isnt getting the image. you should just focus on returning an image if its found or not, not worrying about if the image is tied to an application